File: TerraObserver/TO.Abstractions/Views/Uis/IPlanetHud.cs
using Godot;
using Godot.Abstractions.Bases;

namespace TO.Abstractions.Views.Uis;

/// Copyright (C) 2025 Zhu Xiaohe(aka ZeromaXHe)
/// Author: Zhu XH (ZeromaXHe)
/// Date: 2025-06-22 20:01:22
public interface IPlanetHud : IControl
{
    #region on-ready

    Label CamLonLatLabel { get; }
    PanelContainer CompassPanel { get; }
    TextureRect RectMap { get; }
    LineEdit RadiusLineEdit { get; }
    LineEdit DivisionLineEdit { get; }

    LineEdit ChunkDivisionLineEdit { get; }
    VSlider ElevationVSlider { get; }
    VSlider WaterVSlider { get; }

    #endregion
}

File: TerraObserver/TO.Domains/Types/PlanetHuds/IPlanetHud.fs
namespace TO.Domains.Types.PlanetHuds

open Godot
open Godot.Abstractions.Bases

/// Copyright (C) 2025 Zhu Xiaohe(aka ZeromaXHe)
/// Author: Zhu XH (ZeromaXHe)
/// Date: 2025-06-29 14:13:29
[<Interface>]
[<AllowNullLiteral>] // 因为 PlanetHud 不是 Tool，所以编辑器内可能为空
type IPlanetHud =
    inherit IControl
    // =====【on-ready】=====
    abstract CamLonLatLabel: Label
    abstract CompassPanel: PanelContainer
    abstract RectMap: TextureRect
    abstract RadiusLineEdit: LineEdit
    abstract DivisionLineEdit: LineEdit
    abstract ChunkDivisionLineEdit: LineEdit
    abstract ElevationVSlider: VSlider
    abstract WaterVSlider: VSlider
